samtranslator/model/sam_resources.py

```diff
@@ -886,6 +889,11 @@ def _construct_version(
             # If SnapStart is enabled we want to publish a new version, to have the corresponding snapshot
             if function.SnapStart and function.SnapStart.get("ApplyOn", "None") != "None":
                 logical_dict.update({"SnapStart": function.SnapStart})
+            # We can't directly change AutoPublishAlias as that would be a breaking change, so we have to add this opt-in
+            # property that when set to true would change the lambda version whenever a property in the lambda function changes
+            if self.AutoPublishAliasAllProperties:
```

Comment:
Could this interfere with the `ApplyOn` logic above?

Reply:
Possibly, if they have a function and they add `Snapstart` set to `None` then it would create a new version without the Snapstart being enabled. Should that case be accounted for?

Reply:
Not sure, I don't think it matters. It basically checks that SnapStart is enabled. For `AutoPublishAliasAllProperties` we don't care about any of that I assume.